tests/unbound-method.test.ts

```diff
@@ -14,48 +18,1349 @@ const ruleTester = new RuleTester({
   },
 })
 
+const ConsoleClassAndVariableCode = `
+  class Console {
+    log(str) {
+      process.stdout.write(str);
+    }
+  }
+
+  const console = new Console();
+`.trim()
+
+const toThrowMatchers = [
+  'toThrow',
+  'toThrowError',
+  'toThrowErrorMatchingSnapshot',
+  'toThrowErrorMatchingInlineSnapshot',
+]
+
+const validTestCases: string[] = [
+  ...[
+    'expect(Console.prototype.log).toHaveBeenCalledTimes(1);',
+    'expect(Console.prototype.log).not.toHaveBeenCalled();',
+    'expect(Console.prototype.log).toStrictEqual(somethingElse);',
+    'vitest.mocked(Console.prototype.log).mockImplementation(() => {});',
```

Comment:
Once either #852 or #851 is landed, a new test like this but using `vi` instead of `vitest` should be added